libphal/phal_sbedump.C

```diff
@@ -0,0 +1,228 @@
+extern "C" {
+#include <libgen.h>
+#include <libpdbg.h>
+#include <libpdbg_sbe.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <unistd.h>
+}
+
+#include "libphal.H"
+#include "log.H"
+
+#include <ekb/chips/p10/procedures/hwp/lib/p10_pibmem_dump.H>
+#include <ekb/chips/p10/procedures/hwp/lib/p10_pibms_reg_dump.H>
+#include <ekb/chips/p10/procedures/hwp/lib/p10_ppe_state.H>
+#include <ekb/chips/p10/procedures/hwp/lib/p10_sbe_localreg_dump.H>
+#include <ekb/chips/p10/procedures/hwp/lib/pibms_regs2dump.H>
+#include <ekb/hwpf/fapi2/include/return_code_defs.H>
+#include <fapi2.H>
+
+#include <filesystem>
+#include <fstream>
+#include <string>
+#include <vector>
+
+namespace openpower::phal
+{
+namespace dump
+{
+
+using namespace openpower::phal::logging;
+
+void preCollection(struct pdbg_target* proc)
```

Comment:
Add two utlity function to probe fsi/pib tragets. separately. and place in utils_pdbg.C

Her you can call the fsi/pib probe function.

{
    struct pdbg_target* pibTarget = nullptr;
    pdbg_for_each_target("pib", procTarget, pibTarget)
    {
        // grab first one we find
        break;
    }
    if (!pibTarget)
    {
        log<level::ERR>(
            "pib path of target not found",
            entry("PROC_TARGET_PATH=%s", pdbg_target_path(procTarget)));
       <exception>
    }
    // probe PIB and ensure it's enabled
    if (PDBG_TARGET_ENABLED != pdbg_target_probe(pibTarget))
    {
        log<level::ERR>(
            "probe on pib target failed",
            entry("PIB_TARGET_PATH=%s", pdbg_target_path(pibTarget)));
      <exception>
    }

}